### DISCUSSION TITLE
X-model doesnt update when using safari autofill

### DISCUSSION THREAD

**Discussion by tombien** [alpinejs/alpine#1810] [7. Bugs (archived)] [answered]
Hello, I have
```
<input type="text" x-model="name" name="name" /><span x-text="name"></span>
<script>
    document.addEventListener('alpine:init', () => {
        Alpine.data('orderForm', () => ({
            name: null,
       }));
   });
</script>
```
When I type some value to input field, my model is updated. When I use autofill in Safari, input value is updated but model is not updated (x-text in span is still null).
I tried to put some event log, like this:
```
<input type="text" x-model="name" name="name"
                    @input="console.log(event); console.log(name)"
                    @change="console.log(event); console.log(name)" />
```
When type on keyboard, everything is OK. When using autofill, events are fired, Events have targe.value with correct value, but 'name' is null in console and in span.

Autofill in Chrome is OK and everything is updated.

**Comment by tombien** [comment]
I made a pen for this problem:

https://codepen.io/tombien/pen/ZEKyNxe

**Reply by SimoTod** [reply]
It's a Safari bug/behaviour.
If you add `@input="console.log(event.target.value);` you can see that the value is populated when you normally type or paste a string but it's empty when using the autofill so x-model gets updated with an empty string.

**Reply by tombien** [reply]
1. In Chrome, autofill fires Object Event, in Safari autofill fires Object CustomEvent - is there any difference for Alpine?
2. I am not able to make it in Codepen, but in my app, when I use autofill in Safari, Event.target.value is correct, but model is empty (as you can see in span below input)
![Snímek obrazovky 2021-07-20 v 18 04 46](https://user-images.githubusercontent.com/19426771/126357639-6f78ba6d-3c40-4289-84d4-ddecce36fad1.png)

**Reply by yabdab** [reply]
I am also seeing the value NOT entered in the input field at all when using auto-fill ( in Safari desktop AND mobile ) ?

**Reply by yabdab** [reply]
I can confirm that x-model blocks Safari from inserting values into inputs that contain the x-model tag.

Remove the x-model tag and all works as expected.

This seems to happen when you choose to auto-fill using a contact card. All inputs get the "autofilled" yellow color, but no values are added.

**Comment by MuzafferDede** [comment, marked as answer]
- #1969 should fix this
